src/server/types.ts

```diff
@@ -495,6 +496,10 @@ export interface Plugin<State = Record<string, unknown>> {
   middlewares?: PluginMiddleware<State>[];
 
   islands?: PluginIslands;
+
+  location?: string;
+
+  projectLocation?: string;
```

Comment:
Also, I wonder whether it's worth having `projectLocation?: string | URL`. Doing so would:
1. Allow someone to pass in `Deno.cwd()`, like we do in the SaaSKit PR. We would then use `path.toFileUrl()` from `deno_std`.
2. Not require passing in `url.ref`, but rather just the `url`. We could use `path.toFileUrl()` here too.

WDYT?

Reply:
I like the part about accepting URLs. Although I'm not sure that `Deno.cwd()` is always the right thing. I'll definitely play around with this later today and see what happens.

Reply:
Using `toFileUrl` seems risky when needing to deal with remote plugins.